src/MongoDB.Driver.Core/Core/Compression/ICompressor.cs

```diff
@@ -0,0 +1,60 @@
+﻿/* Copyright 2019–present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.IO;
+using MongoDB.Bson.IO;
+
+namespace MongoDB.Driver.Core.Compression
+{
+    /// <summary>
+    /// Represents the compressor type.
+    /// </summary>
+    public enum CompressorType
+    {
```

Comment:
Might be worth adding a comment like:

        // NOTE: the numeric values of the enum members MUST be kept in sync with the binary wire protocol